apps/aecore/src/aec_chain_state.erl

```diff
@@ -707,7 +707,11 @@ assert_not_illegal_fork_or_orphan(Node, Origin, State) ->
     TopHeight = node_height(Top),
     Height    = node_height(Node),
     case assert_height_delta(Origin, Height, TopHeight) of
-        true -> ok;
+        true ->
+            case assert_eoe(Node) of
+                true -> ok;
+                false -> aec_block_insertion:abort_state_transition(invalid_eoe)
```

Comment:
Is this really the right place for this check? I had guessed somewhere in aec_consensus_hc?

Reply:
There is also the call to this function:
`assert_not_illegal_fork_or_orphan(Node, Origin, State2), %% TODO: move this check to aec_conductor... it doesn't really belong here`
Just moving this to the conductor will not solve the problem that pow will also do this  check.